dbms/src/Interpreters/InterpreterSystemQuery.cpp

```diff
@@ -124,4 +189,66 @@ BlockIO InterpreterSystemQuery::execute()
 }
 
 
+void InterpreterSystemQuery::restartReplicas()
+{
+    for (auto & elem : context.getDatabases())
+    {
+        DatabasePtr & database = elem.second;
+        const String & database_name = elem.first;
+
+        for (auto iterator = database->getIterator(context); iterator->isValid(); iterator->next())
+        {
+            if (!dynamic_cast<const StorageReplicatedMergeTree *>(iterator->table().get()))
+                continue;
+
+            String table_name = iterator->name();
+
+            try
+            {
+                /// Detaching actions
+                {
+                    auto ast_detach = std::make_shared<ASTDropQuery>();
+                    ast_detach->detach = true;
+                    ast_detach->database = database_name;
+                    ast_detach->table = table_name;
+
+                    InterpreterDropQuery interpreter(ast_detach, context);
+                    interpreter.execute();
+                }
+
```

Comment:
Atomicity?
There shouldn't be a moment of time when the client doesn't see that table exists.